kernel/shared/src/main/scala/monocle/function/Index.scala

```diff
@@ -0,0 +1,33 @@
+package monocle.function
+
+import monocle.Optional
+
+import scala.util.Try
+
+trait Index[S] {
```

Comment:
So it is a value of type `A` indxed by  a type `I`? I guess that would be applicable to list like and map like instances 👍

Reply:
indeed, it is even applicable to a String